HashRegistrarSimplified.sol

```diff
@@ -253,7 +253,11 @@ contract Registrar {
      *
      * @param _hash The hash to start an auction on
      */    
-    function startAuction(bytes32 _hash) inState(_hash, Mode.Open) registryOpen() {
+    function startAuction(bytes32 _hash) registryOpen() {
+        var mode = state(_hash);
+        if(mode == Mode.Auction) return;
+        if(mode != Mode.Open) throw;
```

Comment:
Because it's an error to try and open an auction that can't be opened (at least in my mind), and throwing reflects this. If this is being called by `startAuctions`, it also ensures that the call as a whole fails.